distributed/worker.py

```diff
@@ -33,6 +33,8 @@
     typename,
 )
 
+import distributed
+
```

Comment:
instead of this, please use

```
from __future__ import annotations
[...]
from typing import TYPE_CHECKING
if TYPE_CHECKING:
    from . import Client

Reply:
Will this work? There’s a circular import issue, which is why I had to use strings to type it

Reply:
yes, my suggestion is the canonical way of breaking circular imports since python 3.7

Reply:
Ok, thanks! But we still need to wrap the annotations in strings as the resolution is not delayed in the Python versions we support, and it will also break any usage of `typing.get_type_hints` at runtime as `Client` is not defined in this scope.

I'll make the changes, but it seems less than ideal.

Reply:
Resolution is delayed as long as ``from __future__ import annotations`` is at the top of the module.
``typing.get_type_hints`` will return strings when delayed annotations are on, and whatever user code invokes it at runtime has to deal with it to be compliant with PEP 563 (https://www.python.org/dev/peps/pep-0563/)

Reply:
Thanks! I've fixed it 👍